### PR TITLE
Updated library to work with the new asset/dispatch endpoints

### PR DESCRIPTION
**Pull request by mjschuetze102**
## New Features:
- Added additional application management service endpoints
  - GetApplicationInfo - Retrieve detailed information about the application
  - UpdateApplication - Update information about the application 
- Added additional course service endpoints
  - GetCourseAsset - Downloads an asset file from a course
  - GetCourseVersionAsset - Downloads an asset file from a specific course version
  - GetCourseFileList - Gets the file structure of a course
  - GetCourseVersionFileList - Gets the file structure of a specific course version
  - DeleteCourseAsset - Deletes an asset file from a course
  - DeleteCourseVersionAsset - Deletes an asset file from a specific course version 
- Added additional dispatch service endpoints
  - UpdateDispatches - Updates a group of dispatches simultaneously
  - DeleteDispatches - Deletes a group of dispatches simultaneously

## Updates:
- Standardized parameter list for several endpoints. This involved adding additional parameters to several endpoints as well as reordering the parameters
  - GetCourses
  - GetDispatches
  - GetDestinations
  - GetDestinationDispatches
  - GetAllInvitations
  - GetPublicInvitations
  - GetPublicUserInvitations
  - GetPrivateInvitations
  - GetPrivateUserInvitations
  - GetRegistrations
  - GetXapiCredentials
- Changed the schema for several endpoints to only include the values that can be modified
  - CreateApplication
  - UpdatePublicInvitation
  - UpdatePrivateInvitation
- Removed cssUrl parameter from BuildCoursePreviewLaunchLink. The cssUrl in the LaunchLinkRequestSchema remains unchanged
- Updated RestSharp version requirement
- Updated .NET Framework version requirement

### CODE REVIEW

**Review by FishHooks — APPROVED** [review]

**Review by DillGromble — APPROVED** [review]

**Comment by TheConstructor** [comment]
It seems there is a problem with the new NuGet-package: it only contains the readme: https://nuget.info/packages/Com.RusticiSoftware.Cloud.V2/2.0.0
It seems that happened with 1.1.0, too https://nuget.info/packages/Com.RusticiSoftware.Cloud.V2/1.1.0 and 1.1.1 fixed it https://nuget.info/packages/Com.RusticiSoftware.Cloud.V2/1.1.1

File: src/Com.RusticiSoftware.Cloud.V2/Com.RusticiSoftware.Cloud.V2.nuspec [hunk 1]
```diff
@@ -15,28 +15,28 @@
...
             <dependency id="JsonSubTypes" version="1.2.0" />
-            <dependency id="RestSharp" version="105.1.0" />
+            <dependency id="RestSharp" version="106.0.1" />
 
         </dependencies>
     </metadata>
     <files>
         <!-- A readme.txt will be displayed when the package is installed -->
         <file src="..\..\README.md" target="" />
     </files>
```
Comment (TheConstructor) [inline src/Com.RusticiSoftware.Cloud.V2/Com.RusticiSoftware.Cloud.V2.nuspec:41]: It seems the `<files>`-Tag is missing items for the library itself. See e.g. https://docs.microsoft.com/en-us/nuget/reference/nuspec#including-assembly-files

Comment (mjschuetze102) [reply src/Com.RusticiSoftware.Cloud.V2/Com.RusticiSoftware.Cloud.V2.nuspec:41]: Sorry about that. Will be fixing and creating a new version shortly.

Comment (mjschuetze102) [reply src/Com.RusticiSoftware.Cloud.V2/Com.RusticiSoftware.Cloud.V2.nuspec:41]: New version has been published with the changes in #17 to remedy this problem.

**Pull request merged**